scripts/actions/client.js

```diff
@@ -0,0 +1,207 @@
+import { CLIENT_BUSY } from "../constants";
+
+
+export let client;
+
+function getClient({session: {server, username, password}}) {
+  return client || new KintoClient(server, {
+    headers: {
+      Authorization: "Basic " + btoa([username, password].join(":")),
+      // XXX by default we must avoid caching authentication-dependent resources
+      // as they could leak private information to different users. Here we
+      // simply discard caching entirely, waiting for a Kinto server fix.
+      // ref https://github.com/Kinto/kinto/issues/593
+      Pragma: "no-cache",
+    }
+  });
+}
+
+export function resetClient() {
+  client = null;
+}
+
+export function clientBusy(busy) {
+  return {
+    type: CLIENT_BUSY,
+    busy,
+  };
+}
+
+function execute(fn) {
+  return (dispatch, getState) => {
+    dispatch(clientBusy(true));
+    const client = getClient(getState());
+    return fn(client, dispatch, getState)
+      .catch((err) => {
+        dispatch(notifyError(err));
+      })
+      .then(() => {
+        dispatch(clientBusy(false));
+      });
+  };
+}
+
+export function listBuckets() {
+  return execute((client, dispatch, getState) => {
+    return client.fetchServerInfo()
+      .then((serverInfo) => {
+        dispatch(SessionActions.serverInfoLoaded(serverInfo));
+        // XXX We need to first issue a request to the default bucket in order
+        // to create it so we can retrieve the list of buckets.
+        // ref https://github.com/Kinto/kinto/issues/454
+        return client.bucket(serverInfo.user.bucket).getAttributes();
+      })
+      .then(() => {
+        return client.listBuckets();
+      })
+      .then(({data}) => {
+        return Promise.all(data.map((bucket) => {
+          return client.bucket(bucket.id).listCollections()
+            .then(({data}) => ({...bucket, collections: data}));
+        }));
+      })
+      .then((buckets) => {
+        dispatch(SessionActions.bucketListLoaded(buckets));
+      });
+  });
+}
+
+export function createCollection(bid, collectionData) {
+  const {name, schema, uiSchema, displayFields} = collectionData;
+  return execute((client, dispatch, getState) => {
+    return client.bucket(bid).createCollection(name, {
+      data: {uiSchema, schema, displayFields},
+    })
+      .then(({data}) => {
+        dispatch(notifySuccess("Collection created."));
+        dispatch(CollectionActions.collectionCreated(data));
+        dispatch(listBuckets());
+        // XXX redirect to created collection view?
+      });
+  });
+}
+
+export function deleteCollection(bid, cid) {
+  return execute((client, dispatch, getState) => {
+    return client.bucket(bid).deleteCollection(cid)
+      .then(({data}) => {
+        dispatch(notifySuccess("Collection deleted."));
+        dispatch(CollectionActions.collectionDeleted(data));
+        dispatch(listBuckets());
+        dispatch(updatePath(""));
+      });
+  });
+}
+
+export function loadCollectionProperties(bid, cid) {
```

Comment:
There's a useful distinction; the `bid` form means that's a value we've received as a URL parameters, while `bucket` means this is something we received from the server.